src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,59 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
-
+        String line;
+        HashMap<String, Integer> st = new HashMap<>();
+        try (BufferedReader bri = new BufferedReader(new FileReader(fromFileName))) {
+            while ((line = bri.readLine()) != null) {
+                String[] em = line.split(",");
+                if (em.length == 2) {
+                    var tmp1 = em[0].trim();
+                    var tmp2 = em[1].trim();
+                    if (st.containsKey(tmp1)) {
+                        st.replace(tmp1, st.get(tmp1) + Integer.parseInt(tmp2));
+                    } else {
+                        st.put(tmp1, Integer.parseInt(tmp2));
+                    }
+                }
+            }
+        } catch (IOException | NumberFormatException e) {
+            if (e instanceof IOException) {
+                throw new RuntimeException(e.getMessage() + "Problem with input file");
+            } else {
+                throw new RuntimeException(e.getMessage());
+            }
+        }
```

Comment:
Divide program logic

Large methods is very difficult to read and understand. So methods should be short, between 5-15 lines. And it’s important for you to remember that one method should be responsible for one task.

read and write - are a separate operations. So we should create separate methods for them.

Reply:
refactored